### PR TITLE
Feedback implementation

### PR DESCRIPTION
**Pull request by Gabbendorf**

### CODE REVIEW

**Review by RabeaGleissner — COMMENTED** [review]
Looks good!

File: src/main/EchoServer.java [hunk 1]
```diff
@@ -1,26 +1,38 @@
...
+    private void promptToType() {
+        this.output.println("Hi! Please type anything:");
+    }
+
+    private String getWords() {
+        String words = input.nextLine();
         while (words.length() == 0) {
-            System.out.println("Please type at least 1 word:");
-            words = in.nextLine();
+            this.output.println("Please type at least 1 word:");
```
Comment (gemcfadyen) [inline src/main/EchoServer.java]: Usually we use the `this` keyword in the constructor when assigning things, but thereafter in the other methods we don't bother. It is not incorrect, but I would say it is less idiomatic to see `this` used in all methods.

File: src/main/EchoServer.java [hunk 2]
```diff
@@ -1,26 +1,38 @@
...
+            words = input.nextLine();
         }
         return words;
     }
 
-    public void printReversedEcho(String usersWords) {
-        System.out.println(new StringBuilder(usersWords).reverse().toString());
+    private void printReversedEcho(String usersWords) {
+        this.output.println(new StringBuilder(usersWords).reverse().toString());
```
Comment (gemcfadyen) [inline src/main/EchoServer.java]: same as above. If you really want to use `this`, then be consistent and use it on line 30 for `input` as well.

File: src/main/EchoServer.java (hunk 1 above)
Comment (Gabbendorf) [reply src/main/EchoServer.java]: I thought it wouldn't work without reusing 'this' in each method! I tried to remove them from the methods and tests still pass :)

**Pull request merged**